index.js

```diff
@@ -11,31 +11,44 @@ import queueMicrotask from 'queue-microtask'
 /**
  * Parse a torrent identifier (magnet uri, .torrent file, info hash)
  * @param  {string|ArrayBufferView|Object} torrentId
+ * @param  {Object} options
+ * @param  {string} options.hashMode - 'v1', 'v2', or 'both' (default: 'v1')
  * @return {Object}
  */
-async function parseTorrent (torrentId) {
+async function parseTorrent (torrentId, options = {}) {
   if (typeof torrentId === 'string' && /^(stream-)?magnet:/.test(torrentId)) {
     // if magnet uri (string)
     const torrentObj = magnet(torrentId)
 
-    // infoHash won't be defined if a non-bittorrent magnet is passed
-    if (!torrentObj.infoHash) {
+    // infoHash (v1) or infoHashV2 (v2) won't be defined if a non-bittorrent magnet is passed
+    if (!torrentObj.infoHash && !torrentObj.infoHashV2) {
       throw new Error('Invalid torrent identifier')
     }
 
     return torrentObj
   } else if (typeof torrentId === 'string' && (/^[a-f0-9]{40}$/i.test(torrentId) || /^[a-z2-7]{32}$/i.test(torrentId))) {
-    // if info hash (hex/base-32 string)
+    // if info hash v1 (hex/base-32 string)
     return magnet(`magnet:?xt=urn:btih:${torrentId}`)
+  } else if (typeof torrentId === 'string' && /^[a-f0-9]{64}$/i.test(torrentId)) {
+    // if info hash v2 (hex string)
+    return magnet(`magnet:?xt=urn:btmh:1220${torrentId}`)
   } else if (ArrayBuffer.isView(torrentId) && torrentId.length === 20) {
-    // if info hash (buffer)
+    // if info hash v1 (buffer)
     return magnet(`magnet:?xt=urn:btih:${arr2hex(torrentId)}`)
+  } else if (ArrayBuffer.isView(torrentId) && torrentId.length === 32) {
+    // if info hash v2 (buffer)
+    return magnet(`magnet:?xt=urn:btmh:1220${arr2hex(torrentId)}`)
```

Comment:
I'm not exactly sure how the torrentId field is used downstream, but I see ability to specify a torrent by SHA1 hash, SHA256 hash, but I'm not sure how we would pass both in the case of a hybrid torrent here. Would we just treat both swarms as separate? It is legal to have both `btih` and `btmh` in a magnet (and how hybrid magnets usually work)